### DISCUSSION TITLE
provide a method on hd.location that returns the query args as a dictionary

### DISCUSSION THREAD

**Discussion by tkriplean** [hyperdiv/hyperdiv#43] [Ideas]
I'm finding that I'm doing a lot of:

```
    loc = hd.location()
    query_args = dict(
        urllib.parse.parse_qsl(urllib.parse.urlsplit(loc.to_string()).query)
    )
```

I expect that reading query parameters values from a dictionary will be more widely used, as the query parameter string isn't very readily applied.

**Comment by mariusnita** [comment]
Sounds good to provide a method or property that returns a dict.

Btw you can do `dict(urllib.parse.parse_qsl(loc.query_args))`.

**Comment by mariusnita** [comment]
Adding a note: casting the output of `parse_qsl` to `dict()` is incorrect when an arg has multiple values:
```py
urllib.parse.parse_qsl('a=1&b=2&b=3')
> [('a', '1'), ('b', '2'), ('b', '3')]
dict(urllib.parse.parse_qsl('a=1&b=2&b=3'))
> {'a': '1', 'b': '3'}
```

`parse_qs` returns a dict of lists with all the values.
```py
urllib.parse.parse_qs('a=1&b=2&b=3')
{'a': ['1'], 'b': ['2', '3']}
```

**Reply by tkriplean** [reply]
Would be nice if the desire to support multi-valued query parameters (which is more niche) didn't force us to always deal with a length 1 array for single valued parameters.
